core/include/vecmem/memory/unique_ptr.hpp

```diff
@@ -17,6 +17,30 @@
 #include "vecmem/vecmem_core_export.hpp"
 
 namespace vecmem {
+namespace details {
+
+/// Helper function for generating a nonexistent, but still "aligned" pointer.
+///
+/// This is literally just here because MSVC is super touchy about type
+/// casting.
+///
+/// @tparam T The type of the pointer
+/// @return The pointer
+///
+template <typename T>
+T get_nonexistent_pointer() {
+    if constexpr (sizeof(T) == 8) {
+        return reinterpret_cast<T>(0xf000ull);
+    } else if constexpr (sizeof(T) == 4) {
+        return reinterpret_cast<T>(0xf000ul);
+    } else {
+        static_assert(sizeof(T) == 8 || sizeof(T) == 4,
+                      "Unsupported pointer size.");
+    }
+}
```

Comment:
So... MSVC is afraid of casting from a 32-bit value to a 64-bit one. :confused:

https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-1-c4312?view=msvc-170

I'm open to any other solutions. Other than turning off this warning for all clients, this was the cleanest way out of this issue that I could find... :thinking: